### PR TITLE
[Workflow] Added a way to not fire the announce event

### LINKED ISSUE #35197: [Workflow] Hide internal services
Looking at `bin/console debug:container`, there's a lot of noise.

In general im curious if we can prepend service IDs with a dot in 5.0, but im not sure if it involves any BC breaks (or if we care for these services), hence an issue for now.

Example:

```
  state_machine.abstract                  Symfony\Component\Workflow\StateMachine
  state_machine.some                      Symfony\Component\Workflow\StateMachine
  state_machine.some.definition           Symfony\Component\Workflow\Definition 
  state_machine.some.listener.audit_trail Symfony\Component\Workflow\EventListener\AuditTrailListener
  state_machine.some.transition.0         Symfony\Component\Workflow\Transition
```

Having multiple state machines (`some` is a single example), the noise is N*N.

### LINKED ISSUE #35286: [Workflow] ANNOUNCE event should be optional
Q | A
-- | --
Bug report? | no
Feature request? | yes
BC Break report? | no
RFC? | no
Symfony version | 4.4.2

The workflow.announce event has a side effect : it fires guard events for all accessible transitions from the subject's place, even if there is no subscriber registered for this event. AFAIK it's the only event of the workflow component with this behavior.

Code in guard event, usually business rules, can be costly to execute. This cost is useless if the announce event is not leveraged, so the dispatching of this event should be optional.

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix #35286
| License       | MIT
| Doc PR        | https://github.com/symfony/symfony-docs/pull/12942

### CODE REVIEW

File: src/Symfony/Component/Workflow/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ CHANGELOG
 
  * Added context to `TransitionException` and its child classes whenever they are thrown in `Workflow::apply()`
  * Added `Registry::has()` to check if a workflow exists
+ * Added support for `$context[Workflow::DISABLE_ANNONCE_EVENT] = true` when calling `workflow->apply()` to not fire the annonce event
```
Comment (apfelbox) [inline src/Symfony/Component/Workflow/CHANGELOG.md]: I think you have a type throughout the whole PR: `annonce -> announce`

Comment (lyrixx) [reply src/Symfony/Component/Workflow/CHANGELOG.md]: Thanks. Fixed

**Review by noniagriconomie — APPROVED** [review]
Nice one :)

**Review by Harry-Dunne — APPROVED** [review]
Nice!

**Review by OskarStark — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Pull request merged**